beaconChain/tests/utils/state.ts

```diff
@@ -0,0 +1,200 @@
+  latestIndexRoots?: bytes32[];
+  latestPenalizedBalances?: uint64[]; // Balances penalized at every withdrawal period
+  latestAttestations?: PendingAttestation[];
+  batchedBlockRoots?: bytes32[];
+
+  // Ethereum 1.0 deposit root
+  latestEth1Data?: Eth1Data;
+  eth1DataVotes?: Eth1DataVote[];
+}
+
+/**
+ * Generate beaconState, by default it will use the initial state defined when the `ChainStart` log is emitted.
+ * NOTE: All fields can be ovverriden through `opts`.
+ * @param {TestBeaconState} opts
+ * @returns {BeaconState}
+ */
+export function generateState(opts?: TestBeaconState): BeaconState {
+
+  const initialCrosslinkRecord: Crosslink = {
+    epoch: GENESIS_EPOCH,
+    shardBlockRoot: ZERO_HASH
+  };
+
+  return {
+    // MISC
+    slot: opts.slot || GENESIS_SLOT,
+    genesisTime: opts.genesisTime || new Date().getTime(),
+    fork: opts.fork || {
+      previousVersion: GENESIS_FORK_VERSION,
+      currentVersion: GENESIS_FORK_VERSION,
+      epoch: GENESIS_EPOCH
+    },
+    // Validator registry
+    validatorRegistry: opts.validatorRegistry || [],
+    validatorBalances: opts.validatorBalances || [],
+    validatorRegistryUpdateEpoch: opts.validatorRegistryUpdateEpoch || GENESIS_EPOCH,
+
+    // Randomness and committees
+    latestRandaoMixes: opts.latestRandaoMixes || Array.from({length: LATEST_RANDAO_MIXES_LENGTH}, () => ZERO_HASH),
+    previousEpochStartShard: opts.previousEpochStartShard || GENESIS_START_SHARD,
+    currentEpochStartShard: opts.currentEpochStartShard || GENESIS_START_SHARD,
+    previousCalculationEpoch: opts.previousCalculationEpoch || GENESIS_EPOCH,
+    currentCalculationEpoch: opts.currentCalculationEpoch || GENESIS_EPOCH,
+    previousEpochSeed: opts.previousEpochSeed || ZERO_HASH,
+    currentEpochSeed: opts.currentEpochSeed || ZERO_HASH,
+
+    // Finality
+    previousJustifiedEpoch: opts.previousJustifiedEpoch || GENESIS_EPOCH,
+    justifiedEpoch: opts.justifiedEpoch || GENESIS_EPOCH,
+    justificationBitfield: opts.justificationBitfield || 0,
+    finalizedEpoch: opts.finalizedEpoch || GENESIS_EPOCH,
+
+    // Recent state
+    latestCrosslinks: opts.latestCrosslinks || Array.from({length: SHARD_COUNT}, () => initialCrosslinkRecord),
+    latestBlockRoots: opts.latestBlockRoots || Array.from({length: LATEST_BLOCK_ROOTS_LENGTH}, () => ZERO_HASH),
+    latestIndexRoots: opts.latestIndexRoots || Array.from({length: LATEST_INDEX_ROOTS_LENGTH}, () => ZERO_HASH),
+    latestPenalizedBalances: opts.latestPenalizedBalances || Array.from({length: LATEST_PENALIZED_EXIT_LENGTH}, () => 0),
+    latestAttestations: opts.latestAttestations || [],
+    batchedBlockRoots: opts.batchedBlockRoots || [],
+
+    // PoW receipt root
+    latestEth1Data: opts.latestEth1Data || {
+      depositRoot: new Uint8Array(32),
+      blockHash: new Uint8Array(32)
+    },
+    eth1DataVotes: opts.eth1DataVotes || [],
+  };
+}
+
+const randNum = () =>  Math.floor(Math.random() * Math.floor(4));
```

Comment:
If you merge my pull request, I implemented a random number function in utils/misc.ts, so I think it would be better to reuse that here.

Reply:
Hmm good point just looked again you added a upper bound condition. Very useful, check out the gitter I just made a comment on why I haven't merged yet .